src/main/java/com/o19s/es/ltr/stats/suppliers/StoreStatsSupplier.java

```diff
@@ -0,0 +1,138 @@
+        final MultiSearchRequestBuilder requestBuilder = client.prepareMultiSearch();
+        List<String> indices = new ArrayList<>();
+        Stream.of(names)
+                .filter(IndexFeatureStore::isIndexStore)
+                .map(s -> clusterService.state().metadata().index(s))
+                .filter(Objects::nonNull)
+                .map(IndexMetadata::getIndex)
+                .map(Index::getName)
+                .forEach(idx -> {
+                    indices.add(idx);
+                    requestBuilder.add(countSearchRequest(idx));
+                });
+        return createStoreStatsResponse(requestBuilder, indices);
+    }
+
+    private Map<String, Map<String, Object>> createStoreStatsResponse(MultiSearchRequestBuilder requestBuilder,
+                                                                      List<String> indices) {
+        try {
+            MultiSearchResponse msr = requestBuilder.execute().get();
+            assert indices.size() == msr.getResponses().length;
+            Map<String, Map<String, Object>> stats = new ConcurrentHashMap<>(indices.size());
+
+            Iterator<String> indicesItr = indices.iterator();
+            Iterator<MultiSearchResponse.Item> responseItr = msr.iterator();
+            while (indicesItr.hasNext() && responseItr.hasNext()) {
+                MultiSearchResponse.Item it = responseItr.next();
+                String index = indicesItr.next();
+                Map<String, Object> storeStat = initStoreStat(index);
+                stats.put(IndexFeatureStore.storeName(index), storeStat);
+                if (!it.isFailure()) {
+                    Terms aggs = it.getResponse()
+                            .getAggregations()
+                            .get(AGG_FIELD);
+                    aggs.getBuckets()
+                            .stream()
+                            .filter(Objects::nonNull)
+                            .forEach(bucket -> updateCount(bucket, storeStat));
+                }
+            }
+            return stats;
+        } catch (InterruptedException | ExecutionException e) {
+            LOG.error("Error retrieving store stats", e);
+            return Collections.emptyMap();
+        }
+    }
+
+    private Map<String, Object> initStoreStat(String index) {
+        Map<String, Object> storeStat = new HashMap<>();
+        storeStat.put(StatName.STORE_STATUS.getName(), getLtrStoreHealthStatus(index));
+        storeStat.put(StatName.STORE_FEATURE_COUNT.getName(), 0);
+        storeStat.put(StatName.STORE_FEATURE_SET_COUNT.getName(), 0);
+        storeStat.put(StatName.STORE_MODEL_COUNT.getName(), 0);
+        return storeStat;
+    }
+
+    private void updateCount(Terms.Bucket bucket, Map<String, Object> storeStat) {
+        storeStat.computeIfPresent(
+                typeToStatName(bucket.getKeyAsString()),
+                (k, v) -> bucket.getDocCount());
```

Comment:
should you add v? `bucket.getDocCount() + v`
otherwise you might only get the numbers for the last response of you msearch request.

Reply:
Hi @nomoa thanks for the review. I have update this in the latest commit. 

However this should not be a bug IMO. The aggregations will return one term bucket for each type of element, Isn't that correct? In that case, it won't be called more than once per element per store (each store has separate counts map). The initial count for each of these element is 0 and updateCount method is updating this to the actual number of elements in the store. 

Please let me know if I have misunderstood this. I have however updated the code to include